### PR TITLE
Updates: grammar, explanations, status, examples

### PR DESCRIPTION
**Pull request by rwaldron**
Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,39 +1,90 @@
...
+
+## Examples
+
+(The following examples also appear in the README.md of Babel transform plugin for this proposal.)
+
+
+### Decimal Literals
+
+```js
+let budget = 1_000_000_000_000;
```
Comment (ljharb) [inline README.md:20]: Maybe I misunderstood https://github.com/samuelgoto/proposal-numeric-separator/issues/4#issuecomment-304384535 - does "to a single character" mean "only one separator per number", or "each separator can only be a single code unit between digits"?

Comment (rwaldron) [reply README.md:20]: The latter, as in: `1__1` is invalid

Comment (ljharb) [reply README.md:20]: gotcha, thanks for clarifying.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,90 @@
...
-We want to optimize to cover the most common use cases and discourage patterns that would be frowned upon in style guides later on.
 
-With that in mind, here is what we think is a good balance:
+The following grammar represents the Stage 1 criteria, which is: 
 
-- to use the `_` character.
-- only one consecutive underscore is allowed.
-- only between digits (not allowed at the beginning or end of literals).
+1. No leading or trailing separator.
+2. No multiple adjacent separator.
```
Comment (samuelgoto) [inline README.md:83]: Are these equivalent statements? That is, is it isomorphic to say: "only between digits and only one consecutive underscore allowed" and "no leading or trailing, no multiple, no adjancet, etc" (kind of like stating the false negatives or the false positives)?

And if so, which one do you find clearer?

File: README.md [hunk 3]
```diff
@@ -62,43 +113,25 @@ HexDigits ::
...
-
-* Multiple consecutive underscore allowed and only between digits
-* Multiple consecutive underscore allowed, in most positions except for the start of the literal or special positions like a decimal point.
-* Only every other N digits (e.g. N = 3 for decimal literals or 4 for hexadecimal ones)
-
-
 ### Character
 
-More work needs to be done to determine the feasibility and desirability of using different characters. As a reference, most languages use `_` (C++ being the notable exception to use `'`), so `_` is a reasonable starting point.
+The `_` was agreed to as part of Stage 1 acceptance.
```
Comment (samuelgoto) [inline README.md:129]: lgtm

File: README.md [hunk 4]
```diff
@@ -62,43 +113,25 @@ HexDigits ::
...
-Here are some characters that should be looked at to assess feasibility (i.e. is it gramatically possible?) and desirability (e.g. does it lead to a more readable code?):
+The following examples show numeric separators as they appear in other programming languages:
 
-- `_` (Java, Python, Perl, Ruby, Rust, Julia, Ada, C#)
+- **`_` (Java, Python, Perl, Ruby, Rust, Julia, Ada, C#)**
 - `'` (C++)
-- ` `
-- `.`
 
 ## Acknowledgements
```
Comment (samuelgoto) [inline README.md:136]: Mind adding your name to this list?

File: README.md [hunk 5]
```diff
@@ -1,39 +1,90 @@
...
-var pi = 3.14_15; // Instead of 3.1415
-var bytes = 0b11010010_01101001_10010100_10010010; // Instead of 0b11010010011010011001010010010010.
-var 0xCAFE_F00D; // Instead of 0XCAFEF00D.
+// Messages are sent as 24 bit values, but should be 
+// treated as 3 distinct bytes:
+let message = 0xA0_B0_C0;
+
+// What's the value of the upper most byte? It's A0, or 160.
+// We can confirm that:
+let a = (message >> 16) & 0xFF;
```
Comment (samuelgoto) [inline README.md:50]: These are long and unrelated examples. What value do you think they add to this introductory doc?

File: README.md [hunk 6]
```diff
@@ -1,39 +1,90 @@
 # Numeric Separators
 
-This is a [strawman proposal](https://tc39.github.io/process-document/) to introduce a separator in numeric literals in [ECMAScript](https://github.com/tc39/ecma262/).
+## Stage 1
+
+This is a [proposal](https://tc39.github.io/process-document/) to introduce a separator in numeric literals in [ECMAScript](https://github.com/tc39/ecma262/).
```
Comment (samuelgoto) [inline README.md]: Ugh, I may have accepted a PR that will conflict with this. Could you please merge and resolve the conflict?

**Review by samuelgoto — COMMENTED** [review]
These look great, thanks for sending this out!! On a related note, I started kicking off the spec text, want to check it out and provide some guidance?

First time writing one, so apologies for the silly mistakes along the way :)

https://github.com/samuelgoto/proposal-numeric-separator/blob/master/spec.html

File: README.md (hunk 2 above)
Comment (rwaldron) [reply README.md:83]: > Are these equivalent statements? 

They are, I don't remember why I re-wrote them.

File: README.md (hunk 4 above)
Comment (rwaldron) [reply README.md:136]: Sure thing

File: README.md (hunk 5 above)
Comment (rwaldron) [reply README.md:50]: They are very much related and they are long on purpose: they illustrate the visibility/readability advantage that numeric separator offers, via a narrative that programmers can relate to. The readme effectively _is_ the introductory doc—we still need to write the actual spec doc.

**Comment by rwaldron** [comment]
Rebased

File: README.md [hunk 7]
```diff
@@ -1,39 +1,90 @@
 # Numeric Separators
 
-This is a [stage 1 proposal](https://tc39.github.io/process-document/) to introduce a separator in numeric literals in [ECMAScript](https://github.com/tc39/ecma262/).
+## Stage 1
+
+This is a [proposal](https://tc39.github.io/process-document/) to introduce a separator in numeric literals in [ECMAScript](https://github.com/tc39/ecma262/).
```
Comment (samuelgoto) [inline README.md:5]: I'd keep the [stage 1 proposal] bit that was added earlier. Seems constructive to give readers a sense of where things are.

File: README.md [hunk 8]
```diff
@@ -1,39 +1,90 @@
...
+// Let's confirm:
+console.log(budget === 10 ** 12); // true
+```
+
+### Binary Literals
+
+```js
+let nibbles = 0b1010_0001_1000_0101;
+
+// Is bit 7 on? It sure is!
```
Comment (samuelgoto) [inline README.md:33]: Nit: a little weird that "bit 7" refers to the 8th bit :) Fine keeping, just catch my eye as I was reading this through.

File: README.md (hunk 2 above)
Comment (samuelgoto) [reply README.md:83]: Aren't these still equivalent? That is, isn't (a) "No leading or trailing separator." equivalent to "only between digits" (i.e. if it is only between digits, doesn't that imply that leading and trailing are disallowed? and vice versa?), and (b) "No multiple adjacent separator" equivalent to "only one consecutive underscore is allowed"?

File: README.md [hunk 9]
```diff
@@ -62,47 +113,29 @@ HexDigits ::
 
 ### Standard library
 
-We want to make sure that this syntax is consistent with the usage of the standard library. This will involve making libraries like the following compatible with the proposed parsing rules:
+We want to make sure that this syntax is consistent with the usage of the standard library. `Number()` will accept string values that safely convert via `ToNumber()` and ignore occurrences of _NumericLiteralSeparator_. `parseInt()` and `parseFloat()` will remain unchanged.
```
Comment (samuelgoto) [inline README.md:116]: Should we make a note that this leads to a non-backwards-compatible change to Number()?

File: README.md (hunk 2 above)
Comment (rwaldron) [reply README.md:83]: Like I said... 

> They are, I don't remember why I re-wrote them.

Are you trying to subtly tell me that you want me to undo the change?

Comment (samuelgoto) [reply README.md:83]: Ah, nevermind, I was reading this diff incorrectly (I thought both sections were being kept, nevermind, github newbie mistake here).

**Review by samuelgoto — APPROVED** [review]

**Comment by samuelgoto** [comment]
Neat! Thanks @rwaldron !!!

File: README.md (hunk 8 above)
Comment (rwaldron) [reply README.md:33]: 0_o

**Pull request merged**